src/builder.py

```diff
@@ -207,43 +213,145 @@ def run(config: state.BuilderRunConfig, proxy: state.ProxyConfig | None) -> list
     Returns:
         The built image id.
     """
+    build_configs = _parametrize_build(config=config, proxy=proxy)
+    try:
+        with multiprocessing.Pool(len(build_configs)) as pool:
```

Comment:
Not a issue now, but you might want to restrict the upper bound of processes spawned. Similar to how github-runner has a upper bound on runners spawned at once.

Reply:
May I include this as a follow up PR on https://github.com/canonical/github-runner-image-builder-operator/pull/31 please? This would help me reduce the time spent waiting for the integration tests to pass :)
Thank you for the review!

Reply:
Capped number of processes to 10, thank you!